spec/jobs/petition_pdf_generation_worker_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe PetitionPdfGenerationWorker do
+  let(:petition_pdf_service) { spy }
+  let(:worker) { described_class.new petition_pdf_service: petition_pdf_service }
+
+  describe "#perform" do
+    let!(:cycle) { create_cycle_with_phase phases: [{ plugin_type: :petition }] }
+    let!(:phase) { cycle.phases.first }
+    let!(:detail) { create :petition_plugin_detail, plugin_relation: phase.plugin_relation }
+    let!(:version) { create :petition_plugin_detail_version, petition_plugin_detail: detail, published: false }
+
+    let(:document_url) { "https://teste.s3-us-west-2.amazonaws.com/seguranca-publica-peticao-1-1.pdf" }
+
+    subject { worker.perform nil, "{ \"id\": #{version.id} }" }
+
+    before { allow(petition_pdf_service).to receive(:generate).and_return document_url }
+
+    it "generates the pdf and publishs the version" do
+
+      aggregate_failures do
+        subject
+        expect(version.reload.published).to eq true
```

Comment:
Usually in specs like these I avoid hitting the database. I know it is just a couple of tests, but if a lot of other specs do the same, the suite will take a long, long time run.

What a usually do when possible is to mock the things.

Because you are not using the repository pattern, you are going to have to mock `PetitionPlugin::DetailVersion.find`, and just spy on your `version` spy that the update was called with the right input.

The bellow test about the record not found would not be necessary, because of the mock (would not make sense).